### DISCUSSION TITLE
Is it possible to use a custom input component?

### DISCUSSION THREAD

**Discussion by mdegrees** [pacocoursey/cmdk#53] [Q&A]
I'm trying to use an editor (Slate.js) as the input component. I did visually hide <Command.input/> and started controlling it with Slate value but it feels hacky and might compromise accessibility. Is there a better way? 
Thank you for the awesome work!

**Comment by pacocoursey** [comment]
Hmm yeah that's tough. I think for a11y purposes the combobox should still have an `<input />` element, but you don't have to necessarily show it, so your approach is probably solid. I'm guessing you want more complex things rendered in the input, like suggestions or autocomplete?

**Reply by joaom00** [reply]
I think the input component could have the [`asChild`](https://www.radix-ui.com/docs/primitives/utilities/slot) prop like in the radix components

**Reply by mdegrees** [reply]
@pacocoursey Actually it's more about using fields within the input element and showing different commands as the user fills in the fields, selects,... . Slate makes a lot of sense for this use case.
